shell/browser/notifications/mac/notification_center_delegate.mm

```diff
@@ -6,63 +6,151 @@
+      interfaceWithProtocol:@protocol(NSAlertResponceXPCProtocol)];
+  connectionToService_.exportedObject = self;
+  [connectionToService_ resume];
+  NSLog(@"connectionToService_ : %@", connectionToService_);
+  NSLog(@"connectionToService_.remoteObjectProxy : %@",
+        connectionToService_.remoteObjectProxy);
+  NSLog(@"connectionToService_.exportedObject    : %@",
+        connectionToService_.exportedObject);
+  connection_.reset(connectionToService_);
   presenter_ = presenter;
   return self;
 }
 
-- (void)userNotificationCenter:(NSUserNotificationCenter*)center
-        didDeliverNotification:(NSUserNotification*)notif {
-  auto* notification = presenter_->GetNotification(notif);
-  if (notification)
-    notification->NotificationDisplayed();
+- (void)didDeliverNotification:(NSUserNotification*)notif {
+  NSLog(@"didDeliverNotification");
+  dispatch_async(dispatch_get_main_queue(), ^{
+    auto* notification = presenter_->GetNotification(notif);
+    if (notification)
+      notification->NotificationDisplayed();
+  });
 }
 
-- (void)userNotificationCenter:(NSUserNotificationCenter*)center
-       didActivateNotification:(NSUserNotification*)notif {
-  auto* notification = presenter_->GetNotification(notif);
+- (void)didActivateNotification:(NSUserNotification*)notif {
+  BOOL isMainThread = [NSThread isMainThread];
+  NSLog(@"didActivateNotification (is MainThread -> %@): %@",
+        isMainThread ? @"YES" : @"NO", notif);
+  dispatch_async(dispatch_get_main_queue(), ^{
+    // SAP-14881 - Persistent notification on macOS
+    // code bellow provides support for handling notifications from cold start
+    auto* app = electron::api::App::Get();
+    if (app) {
+      const char* action =
+          notif.additionalActivationAction == nil
+              ? nullptr
+              : [notif.additionalActivationAction.identifier UTF8String];
+      const char* reply =
+          notif.response == nil ? nullptr : [notif.response.string UTF8String];
+      // SAP-15908 refine notification-activation for cold start in macOS
+      if (!action && !reply) {  // click on toast body
+        // in case if data exists forwarding it as invoked argument
+        NSString* _data = notif.userInfo ? notif.userInfo[@"_data"] : @"";
+        action = [_data UTF8String];
+      }
+      // SAP-15908 - Refine notification-activation for cold start in macOS
+      std::stringstream stm;
+      if (reply) {
+        NSString* _replyKey =
+            notif.userInfo ? notif.userInfo[@"_replyKey"] : @"";
+        const char* key = [_replyKey UTF8String];
+        stm << L"[";  // json array open brace
+        stm << L"{";  // open brace for key-value pair
+        stm << L"\"" << (key ? key : "") << L"\"";
+        stm << L":";  // json delimeter (i.e colon) between key value
+        stm << L"\"" << reply << L"\"";
+        stm << L"}";  // close brace for key-value pair
+        stm << L"]";  // json array close brace
```

Comment:
Please consider my explanations above https://github.com/electron/electron/pull/34671#discussion_r903013840
Things with performance are playing significant role in case of hundred's or thousands notifications in few seconds

Reply:
As a middle ground, we might consider using [Chromium's JSON writer](https://source.chromium.org/chromium/chromium/src/+/main:base/json/json_writer.h?q=base%2Fjson%2Fjson_writer.h&ss=chromium) cc @MarshallOfSound

Reply:
> Things with performance are playing significant role in case of hundred's or thousands notifications in few seconds

I can't imagine a valid usecase for making thousands of notifications in a few seconds.  But even if there was the cost of constructing objects / using a real JSON writer wouldn't be impactful enough to cause perf issues IMO

Reply:
Hello @MarshallOfSound  regarding to Your words _using a real JSON writer_ - to be honest I do not prefer +1 dependency in case when it can be done into more simple way (the motto of current implementation is : better simpler than clever) 

_I can't imagine a valid usecase for making thousands of notifications in a few seconds_ -  this scenario can be real especially in case of usage renotify. I would propose to imagine notification which present for specific trading chart behavior. Modern stock exchanges can be too volatile regarding to market data feed. What if notification is used as signal to trader?